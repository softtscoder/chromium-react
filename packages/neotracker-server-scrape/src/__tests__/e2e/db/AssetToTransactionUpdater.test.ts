import {
  AssetToTransactionRevert,
  AssetToTransactionSave,
  AssetToTransactionUpdater,
} from '../../../db/AssetToTransactionUpdater';
import { updaterUnitTest, UpdaterUtilTestOptions } from '../../data/updaterTestUtil';

const savesInit: ReadonlyArray<{
  readonly assetIDs: ReadonlyArray<string>;
  readonly transactionID: string;
}> = [
  {
    assetIDs: ['OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO', 'tttttttttttttttttttttttttttttttttttttttt'],
    transactionID: '123',
  },
];

const savesSecondary: ReadonlyArray<{
  readonly assetIDs: ReadonlyArray<string>;
  readonly transactionID: string;
}> = [
  {
    assetIDs: ['cccccccccccccccccccccccccccccccccccccccc', 'dddddddddddddddddddddddddddddddddddddddd'],
    transactionID: '234',
  },
];

const getInputs = (): AssetToTransactionSave => ({
  transactions: savesInit,
});

const getSecondaryInputs = (): AssetToTransactionSave => ({
  transactions: savesSecondary,
});

const getReversions = () => ({
  transactionIDs: ['234'],
});

const utilOptions: UpdaterUtilTestOptions<AssetToTransactionSave, AssetToTransactionRevert, undefined> = {
  name: `AssetToTransaction Updater`,
  createUpdater: () => new AssetToTransactionUpdater(),
  getInputs,
  getSecondaryInputs,
  getReversions,
};

updaterUnitTest(utilOptions);
