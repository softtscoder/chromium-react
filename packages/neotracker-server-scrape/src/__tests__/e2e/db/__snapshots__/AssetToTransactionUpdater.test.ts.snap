// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetToTransaction Updater AssetToTransaction Updater: handles duplicate inserts/saves 1`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [
    Object {
      "id1": "OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
      "id2": "123",
    },
    Object {
      "id1": "tttttttttttttttttttttttttttttttttttttttt",
      "id2": "123",
    },
  ],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AssetToTransaction Updater AssetToTransaction Updater: handles insert/save 1`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [
    Object {
      "id1": "OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
      "id2": "123",
    },
    Object {
      "id1": "tttttttttttttttttttttttttttttttttttttttt",
      "id2": "123",
    },
  ],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AssetToTransaction Updater AssetToTransaction Updater: handles reverting inputs/saves 1`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [
    Object {
      "id1": "OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
      "id2": "123",
    },
    Object {
      "id1": "tttttttttttttttttttttttttttttttttttttttt",
      "id2": "123",
    },
  ],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AssetToTransaction Updater AssetToTransaction Updater: handles reverting inputs/saves 2`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [
    Object {
      "id1": "OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
      "id2": "123",
    },
    Object {
      "id1": "tttttttttttttttttttttttttttttttttttttttt",
      "id2": "123",
    },
    Object {
      "id1": "cccccccccccccccccccccccccccccccccccccccc",
      "id2": "234",
    },
    Object {
      "id1": "dddddddddddddddddddddddddddddddddddddddd",
      "id2": "234",
    },
  ],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;

exports[`AssetToTransaction Updater AssetToTransaction Updater: handles reverting inputs/saves 3`] = `
Object {
  "action": Array [],
  "address": Array [],
  "addressToTransaction": Array [],
  "addressToTransfer": Array [],
  "asset": Array [],
  "assetToTransaction": Array [
    Object {
      "id1": "OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO",
      "id2": "123",
    },
    Object {
      "id1": "tttttttttttttttttttttttttttttttttttttttt",
      "id2": "123",
    },
  ],
  "block": Array [],
  "coin": Array [],
  "contract": Array [],
  "dataPoint": Array [],
  "migration": Array [],
  "processedIndex": Array [],
  "transaction": Array [],
  "transactionInputOutput": Array [],
  "transfer": Array [],
}
`;
